Framework/Crystal/test/PeakStatisticsToolsTest.h

```diff
@@ -0,0 +1,356 @@
+public:
+  explicit MockUniqueReflectionCollection(
+      const std::map<V3D, UniqueReflection> &reflections,
+      const PointGroup_sptr &pointGroup =
+          PointGroupFactory::Instance().createPointGroup("1"))
+      : UniqueReflectionCollection(reflections, pointGroup) {}
+};
+}
+
+class PeakStatisticsToolsTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static PeakStatisticsToolsTest *createSuite() {
+    return new PeakStatisticsToolsTest();
+  }
+  static void destroySuite(PeakStatisticsToolsTest *suite) { delete suite; }
+
+  void test_UniqueReflectionsConstructor() {
+    V3D hkl(1, 1, 1);
+    UniqueReflection reflection(hkl);
+
+    TSM_ASSERT_EQUALS("Constructed UniqueReflection does not have 0 peaks.",
+                      reflection.count(), 0);
+    TSM_ASSERT_EQUALS(
+        "HKL is not equal to constructor argument in UniqueReflection",
+        reflection.getHKL(), hkl);
+  }
+
+  void test_UniqueReflectionsPeaks() {
+    UniqueReflection reflection(V3D(2, 3, 4));
+
+    Peak peak;
+    TS_ASSERT_THROWS_NOTHING(reflection.addPeak(peak));
+    TSM_ASSERT_EQUALS("UniqueReflection count is not 1 after adding peak.",
+                      reflection.count(), 1);
+    TSM_ASSERT_EQUALS(
+        "UniqueReflection peaks vector size is not 1 after adding peak.",
+        reflection.getPeaks().size(), 1);
+  }
+
+  void test_UniqueReflectionsGetIntensitiesAndSigmas() {
+    UniqueReflection reflection(V3D(2, 3, 4));
+
+    std::vector<Peak> peaks = getPeaksWithIandSigma({30.0, 34.0}, {4.5, 6.5});
+    for (auto peak : peaks) {
+      reflection.addPeak(peak);
+    }
+
+    std::vector<double> intensities = reflection.getIntensities();
+    TSM_ASSERT_EQUALS("Intensity vector from UniqueReflection has wrong size.",
+                      intensities.size(), 2);
+    TS_ASSERT_EQUALS(intensities[0], 30.0);
+    TS_ASSERT_EQUALS(intensities[1], 34.0);
+
+    std::vector<double> sigmas = reflection.getSigmas();
+    TSM_ASSERT_EQUALS("Sigma vector from UniqueReflection has wrong size.",
+                      sigmas.size(), 2);
+    TS_ASSERT_EQUALS(sigmas[0], 4.5);
+    TS_ASSERT_EQUALS(sigmas[1], 6.5);
+  }
+
+  void test_UniqueReflectionRemoveOutliersSigmaCrit() {
+    UniqueReflection reflection(V3D(2, 3, 4));
+    TS_ASSERT_THROWS_NOTHING(reflection.removeOutliers(3.0));
+    TS_ASSERT_THROWS(reflection.removeOutliers(0.0), std::invalid_argument);
+    TS_ASSERT_THROWS(reflection.removeOutliers(-10.0), std::invalid_argument);
+  }
+
+  void test_UniqueReflectionRemoveOutliersFewPeaks() {
+    std::vector<Peak> peaks = getPeaksWithIandSigma({30.0, 34.0}, {4.5, 6.5});
+
+    UniqueReflection reflection(V3D(2, 3, 4));
+    reflection.addPeak(peaks[0]);
+
+    TS_ASSERT_THROWS_NOTHING(reflection.removeOutliers());
```

Comment:
I just realized this doesn't really test anything anymore because the reflection is not modified by `removeOutliers`.